subsys/bluetooth/adv_prov/providers/fast_pair.c

```diff
@@ -19,10 +22,26 @@ static enum bt_fast_pair_adv_battery_mode adv_battery_mode = (
 							BT_FAST_PAIR_ADV_BATTERY_MODE_HIDE_UI_IND)
 );
 
+static struct bt_data gen_ad;
+static struct bt_fast_pair_adv_config gen_adv_config = {
+	.adv_mode = BT_FAST_PAIR_ADV_MODE_COUNT,
+	.adv_battery_mode = BT_FAST_PAIR_ADV_BATTERY_MODE_COUNT,
+};
+
+
+static void generated_adv_payload_invalidate(void)
+{
+	gen_adv_config.adv_mode = BT_FAST_PAIR_ADV_MODE_COUNT;
+	gen_adv_config.adv_battery_mode = BT_FAST_PAIR_ADV_BATTERY_MODE_COUNT;
+}
 
 void bt_le_adv_prov_fast_pair_enable(bool enable)
 {
 	enabled = enable;
+
+	if (!enabled) {
+		generated_adv_payload_invalidate();
+	}
```

Comment:
I wonder if we should actually invalidate the payload here or wait until we are requested to provide data and then invalidate the advertising payload. I think that moving payload invalidation could prevent some edge cases (e.g. related to avoiding providing the discoverable advertising payload when RPA was rotated during ongoing advertising session).